third_party/chimp/include/ring_buffer.hpp

```diff
@@ -0,0 +1,69 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// third_party/chimp/include/ring_buffer.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cstring>
+
+namespace duckdb_chimp {
+
+class RingBuffer {
+public:
+	static constexpr uint8_t RING_SIZE = 128;
+	static constexpr uint64_t LEAST_SIGNIFICANT_BIT_MASK = ((uint64_t)1 << (6 + 7 + 1)) - 1;
+	//this.indices = new int[(int) Math.pow(2, threshold + 1)];
+	//! Since threshold is now always set to (6 + 7), we can hardcode this to
+	static constexpr uint16_t INDICES_SIZE = 1 << (6 + 7 + 1); //16384
```

Comment:
Threshold isn't hardcoded to 13, FLOAT has a threshold of 12, use the ThresholdValue<T>::value here as well